designdocs/logical_batching.md

```diff
@@ -0,0 +1,67 @@
+# Summary
+
+In other documents, we explore ways to structure an explicit batch of messages.
+That is, in terms of concrete structure. This document has a different aim --
+deriving the idea of a batch logically, as a product of fields present on other
+entities in our system.
+
+## Background
+
+We are currently experimenting with a new system architecture that has the
+following entities:
+
+1. Schema
+   - Id
+   - Format
+   - FormatType
+   - PayloadLocation
+2. Message
+   - Off-chain / On-chain Payload
+
+*Protocols* would be responsible for indicating not only the shape of the
+underlying payload, but also its format and location. This is important, because
+a protocol's format property would indicate whether a given payload contains a
+batch or not.
+
+## FormatType
+
+Payloads can either be of format type `PARQUET` or `AVRO`. These file formats
+indicate specifically whether their payloads are singular or arrays of objects
+(More info needed here).
+
+## PayloadLocation
+
+The combination of format and location constrain possible payload types:
```

Comment:
This could use a little more elaboration.

Reply:
Agreed, I'm not sure what else to say here though. I'm open to suggestions.